src/movies.js

```diff
@@ -1,16 +1,56 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+
+function getAllDirectors(movies) {
+  const listOfDirector = movies.map(function (arr) {
+    return arr.director;
+  });
+  
+  console.log(listOfDirector);
+}
+
+getAllDirectors(movies)
+
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(key) {
+  const dramaSP = movies.filter(function (key2) {
+    return key2.director === 'Steven Spielberg';
+  });
+
+  console.log(dramaSP);
+}
+
+howManyMovies(movies);
+
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+
+function scoresAverage(key) {
+  const total = key.reduce(function (acc, currentValue) {
+    return acc + currentValue.score;
+  }, 0);
+  return total / key.length;
+}
+
+console.log(scoresAverage(movies));
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+
+function scoresAverage(key) {
+  const total = key.reduce(function (acc, currentValue) {
+    if (movies.genre === 'Drama') {
+      return acc + currentValue.score;
+    }
+  }, 0);
+  return total / key.length;
+}
```

Comment:
1. Is that correct name for the function? ;) You are checking drama movies score here...
2. what is 'movies' here on line 46? This variable is not defined in your function.. Instead of that you should be accessing the currentValue .genre
3. Always analyze the data you are working with, if you look at data.js file, you can see that genre in your object is actually an array. So instead of checking if array is equal to a string, you should be checking if array includes that string